src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -84,6 +84,13 @@ public sealed class CompletionItem : IComparable<CompletionItem>
         /// </summary>
         internal string ProviderName { get; set; }
 
+        /// <summary>
+        /// Indicate whether this <see cref="CompletionItem"/> is cached and reused across completion sessions. 
+        /// This might be used by completion system for things like deciding whether it can safaly cache and reuse
+        /// other data correspodning to this item.
+        /// </summary>
+        internal bool IsCached { get; set; }
```

Comment:
probably want to have an tracking issue on how to address this in long term?

Reply:
and put the link in the comment?

Reply:
Done

Reply:
actually, this is immutable data for this roslyn completion item for importing completion right? then... you could just put this in immutable property? since it won't change once completion item is created?